pkg/registry/core/pod/storage/storage.go

```diff
@@ -245,10 +244,6 @@ func (r *BindingREST) setPodHostAndAnnotations(ctx context.Context, podUID types
 		for k, v := range annotations {
 			pod.Annotations[k] = v
 		}
-		podutil.UpdatePodCondition(&pod.Status, &api.PodCondition{
-			Type:   api.PodScheduled,
-			Status: api.ConditionTrue,
-		})
```

Comment:
This code was explicitly added by https://github.com/kubernetes/kubernetes/pull/24459. Please make sure the reason for adding doesn't exist/make sense anymore.